realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp

```diff
@@ -1503,6 +1896,8 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeIsNotNull(JNIEnv*
                 case type_Float:
                 case type_Double:
                 case type_Timestamp:
+                case type_Decimal:
+                case type_ObjectId:
                     pQuery->not_equal(ColKey(column_idx), realm::null());
                     break;
                 default:
```

Comment:
Unrelated to the PR, I just happened to notice: Why are we having "default:" here (and many other places) - it's better not to to ensure the compiler throws a warning when we are missing the handling of a new enum.

Reply:
Historically this (i.e `REALM_UNREACHABLE`) has been used by Core, it could be interesting in some use case to use an extensive switch statement to ensure we're covering all branches, which in this example will be a fall through for `'type_OldTable', 'type_OldMixed', and 'type_OldDateTime`